src/app/qgstipgui.cpp

```diff
@@ -24,11 +24,11 @@
 #include <qgstipfactory.h>
 
 #ifdef Q_OS_MACX
-QgsTipGui::QgsTipGui()
-    : QDialog( NULL, Qt::WindowSystemMenuHint )  // Modeless dialog with close button only
```

Comment:
I changed the comment because the dialog is shown via its ```exec()``` method which - according to [the documentation](http://doc.qt.io/qt-4.8/qdialog.html#exec) - always makes the dialog modal, even on Mac.